src/test/java/com/louis/HashTableTest.java

```diff
@@ -0,0 +1,112 @@
+package com.louis;
+
+import static org.junit.Assert.assertEquals;
+import org.junit.Test;
+import org.junit.After;
+import org.junit.AfterClass;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import java.util.ArrayList;
+import java.util.HashSet;
+
+import org.junit.rules.ExpectedException;
+import org.junit.Rule;
+
+/**
+ * Tests for {@link HashTable}.
+ * 
+ * @author Louis Ashton  (louisashton@live.com)
+ */
+public class HashTableTest {
+    
+    @Rule
+    public final ExpectedException exception = ExpectedException.none();
+    
+    HashTable listIntegerTable = new HashTable();
+    HashTable stringListTable = new HashTable();
+    
+    /**
+     * Tests construction of hashtables.
+     * 
+     * Checks whether hashtables of various types can be constructed.
+     */
+    @Test
+    public void canConstructHashTables() {
+        assertEquals(listIntegerTable.size(), 0);
+        assertEquals(stringListTable.size(), 0);
+    }
+    
+    /**
+     * Tests illegal argument exceptions.
+     * 
+     * Checks whether hashtables throw IllegalArgumentException when passed null values.
+     */
+    @Test
+    public void cannotGetNull() {
+        exception.expect(IllegalArgumentException.class);
+        stringListTable.get(null);
+    }
+
+    @Test
+    public void cannotPutNull() {
+        exception.expect(IllegalArgumentException.class);
+        stringListTable.put("k",null);
+    }
+    
+    /**
+     * Tests addition to hashtables.
+     * 
+     * Checks whether entries can be put into the hashtables.
+     */
+    @Test
+    public void canPutEntries() {
+        for (int i = 0; i < 1000; i++) {
```

Comment:
Test the resizing separately from the other things. Try and test one concept only in each test.